tools/firmware-utils/src/tplink-safeloader.c

```diff
@@ -1298,6 +1298,35 @@ static struct device_info boards[] = {
 		.last_sysupgrade_partition = "file-system"
 	},
 
+	/** Firmware layout for the EAP225-Wall v2 */
+	{
+		.id     = "EAP225-WALL-V2",
+		.support_list =
+			"SupportList:\r\n"
+			"EAP225-Wall(TP-Link|UN|AC1200-D):2.0\r\n",
+		.support_trail = '\xff',
+		.soft_ver = NULL,
+		.soft_ver_compat_level = 1,
+
+		.partitions = {
+			{"fs-uboot", 0x00000, 0x20000},
+			{"partition-table", 0x20000, 0x02000},
+			{"default-mac", 0x30000, 0x01000},
+			{"support-list", 0x31000, 0x00100},
+			{"product-info", 0x31100, 0x00400},
+			{"soft-version", 0x32000, 0x00100},
+			{"firmware", 0x40000, 0xd80000},
+			{"user-config", 0xdc0000, 0x30000},
```

Comment:
This doesn't match the partitions in DTS. Here, you have a gap from 0xe00000 to 0xf30000, and in partitions you have the config partition at a different location?

Reply:
I'm afraid this is one of those TP-Link screw-ups. I've put the config partition in the DTS at the location determined by the original bootargs. I could correct the partition table here, but this is the way it's described in stock.

From FW v1.20.0:
```
partition fs-uboot base 0x00000 size 0x20000
partition partition-table base 0x20000 size 0x02000
partition default-mac base 0x30000 size 0x01000
partition support-list base 0x31000 size 0x00100
partition product-info base 0x31100 size 0x00400
partition soft-version base 0x32000 size 0x00100
partition os-image base 0x40000 size 0x180000
partition file-system base 0x1c0000 size 0xc00000
partition user-config base 0xdc0000 size 0x30000
partition mutil-log base 0xf30000 size 0x80000
partition oops base 0xfb0000 size 0x40000
partition radio base 0xff0000 size 0x10000
```
Bootargs:
```
mtdparts=spi0.0:
    128k(u-boot),
    64k(pation-table),
    64k(product-info),
    1536k(kernel),
    13568k(rootfs),
    192k(config),
    512k(mutil-log),
    256k(oops),
    64k(ART)
```

Boot log:
```
[    0.470000] Creating 9 MTD partitions on "spi0.0":
[    0.480000] 0x000000000000-0x000000020000 : "u-boot"
[    0.490000] 0x000000020000-0x000000030000 : "pation-table"
[    0.490000] 0x000000030000-0x000000040000 : "product-info"
[    0.500000] 0x000000040000-0x0000001c0000 : "kernel"
[    0.500000] 0x0000001c0000-0x000000f00000 : "rootfs"
[    0.510000] mtd: partition "rootfs" set to be root filesystem
[    0.520000] 0x000000f00000-0x000000f30000 : "config"
[    0.520000] 0x000000f30000-0x000000fb0000 : "mutil-log"
[    0.530000] 0x000000fb0000-0x000000ff0000 : "oops"
[    0.540000] 0x000000ff0000-0x000001000000 : "ART"
```

Reply:
How nice...

Is there actually config data in the marked partition (if this can be determined at all)?

I don't have a strong opinion here, but having inconsistent safeloader and partitions seems undesirable to me. I assume the config partition still is recreated anyway if overwritten like with older devices?

Reply:
Yes, on a 16M device I'd also prefer to be safe and keep the partition as suggested.

Reply:
>  I had dumped the the NAND

I thought this is SPI-NOR?

Reply:
Sorry, my mistake. It's SPI-NOR.